Sources/KSCrashRecordingCore/KSDate.c

```diff
@@ -25,32 +25,48 @@
 #include "KSDate.h"
 
 #include <stdio.h>
+#include <string.h>
 #include <sys/time.h>
 #include <time.h>
 
-void ksdate_utcStringFromTimestamp(time_t timestamp, char *buffer21Chars)
+void ksdate_utcStringFromTimestamp(time_t timestamp, char *buffer, size_t bufferSize)
 {
+    memset(buffer, 0, bufferSize);
+
+    // Cannot be in the future.
+    if (timestamp > ksdate_seconds()) {
+        return;
+    }
+
     struct tm result = { 0 };
-    gmtime_r(&timestamp, &result);
-    snprintf(buffer21Chars, 21, "%04d-%02d-%02dT%02d:%02d:%02dZ", result.tm_year + 1900, result.tm_mon + 1,
+    if (!gmtime_r(&timestamp, &result)) {
+        return;
+    }
+    snprintf(buffer, bufferSize, "%04d-%02d-%02dT%02d:%02d:%02dZ", result.tm_year + 1900, result.tm_mon + 1,
```

Comment:
These format strings can easily overflow which is why I introduced the buffer size macro in the header and pass in the size.